File: backend/src/main/java/com/lawfirm/domain/hr/entity/Regularization.java
package com.lawfirm.domain.hr.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.lawfirm.common.base.BaseEntity;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;

/**
 * 转正申请实体
 */
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("hr_regularization")
public class Regularization extends BaseEntity {

    /**
     * 员工ID
     */
    private Long employeeId;

    /**
     * 申请编号
     */
    private String applicationNo;

    /**
     * 试用期开始日期
     */
    private LocalDate probationStartDate;

    /**
     * 试用期结束日期
     */
    private LocalDate probationEndDate;

    /**
     * 申请日期
     */
    private LocalDate applicationDate;

    /**
     * 预计转正日期
     */
    private LocalDate expectedRegularDate;

    /**
     * 自我评价
     */
    private String selfEvaluation;

    /**
     * 上级评价
     */
    private String supervisorEvaluation;

    /**
     * HR评价
     */
    private String hrEvaluation;

    /**
     * 状态：PENDING-待审批, APPROVED-已通过, REJECTED-已拒绝
     */
    private String status;

    /**
     * 审批人ID
     */
    private Long approverId;

    /**
     * 审批日期
     */
    private LocalDate approvedDate;

    /**
     * 审批意见
     */
    private String comment;
}

File: frontend/apps/web-antd/src/api/hr/regularization.ts
/**
 * 转正申请管理 API
 */
import { requestClient } from '#/api/request';

import type { PageResult } from '../matter/types';

// ========== 类型定义 ==========
export interface RegularizationDTO {
  id: number;
  employeeId: number;
  userId?: number;
  employeeName?: string;
  applicationNo?: string;
  probationStartDate?: string;
  probationEndDate?: string;
  applicationDate?: string;
  expectedRegularDate?: string;
  selfEvaluation?: string;
  supervisorEvaluation?: string;
  hrEvaluation?: string;
  status?: string;
  statusName?: string;
  approverId?: number;
  approverName?: string;
  approvedDate?: string;
  comment?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface RegularizationQuery {
  pageNum?: number;
  pageSize?: number;
  keyword?: string;
  status?: string;
  employeeId?: number;
  departmentId?: number;
}

export interface CreateRegularizationCommand {
  employeeId: number;
  probationStartDate?: string;
  probationEndDate?: string;
  expectedRegularDate?: string;
  selfEvaluation?: string;
}

export interface ApproveRegularizationCommand {
  approved: boolean;
  comment?: string;
}

// ========== API 函数 ==========

/** 分页查询转正申请 */
export function getRegularizationList(params: RegularizationQuery) {
  return requestClient.get<PageResult<RegularizationDTO>>('/hr/regularization', { params });
}

/** 根据ID查询转正申请 */
export function getRegularizationDetail(id: number) {
  return requestClient.get<RegularizationDTO>(`/hr/regularization/${id}`);
}

/** 根据员工ID查询转正申请 */
export function getRegularizationsByEmployeeId(employeeId: number) {
  return requestClient.get<RegularizationDTO[]>(`/hr/regularization/employee/${employeeId}`);
}

/** 创建转正申请 */
export function createRegularization(data: CreateRegularizationCommand) {
  return requestClient.post<RegularizationDTO>('/hr/regularization', data);
}

/** 审批转正申请 */
export function approveRegularization(id: number, data: ApproveRegularizationCommand) {
  return requestClient.post<RegularizationDTO>(`/hr/regularization/${id}/approve`, data);
}

/** 删除转正申请 */
export function deleteRegularization(id: number) {
  return requestClient.delete(`/hr/regularization/${id}`);
}

